src/dsc.h

```diff
@@ -486,55 +499,21 @@ class IdList {
         return t;
     }
 
-    int IndexOf(H h) {
-        if(IsEmpty()) {
-            return -1;
-        }
-        auto it = LowerBound(h);
-        auto idx = std::distance(begin(), it);
-        if (idx < n) {
-            return idx;
-        }
-        return -1;
-    }
-
     T *FindByIdNoOops(H h) {
-        if(IsEmpty()) {
-            return nullptr;
-        }
         auto it = LowerBound(h);
-        if (it == nullptr || it == end()) {
-            return nullptr;
-        }
-        if (it->h.v == h.v) {
-            return it;
+        if(it != end() && it->h.v == h.v) {
+            return &*it;
         }
         return nullptr;
     }
 
-    T *First() {
-        return (IsEmpty()) ? NULL : &(elem[0]);
-    }
-    T *Last() {
-        return (IsEmpty()) ? NULL : &(elem[n-1]);
-    }
-    T *NextAfter(T *prev) {
-        if(IsEmpty() || !prev) return NULL;
-        if(prev - First() == (n - 1)) return NULL;
-        return prev + 1;
-    }
-
-    T &Get(size_t i) { return elem[i]; }
-    T const &Get(size_t i) const { return elem[i]; }
+    T &Get(size_t i) { return elem[indexes[i]]; }
+    T const &Get(size_t i) const { return elem[indexes[i]]; }
     T &operator[](size_t i) { return Get(i); }
     T const &operator[](size_t i) const { return Get(i); }
 
-    T *begin() { return IsEmpty() ? nullptr : &elem[0]; }
-    T *end() { return IsEmpty() ? nullptr : &elem[0] + n; }
-    const T *begin() const { return IsEmpty() ? nullptr : &elem[0]; }
```

Comment:
As above - the const iterators may be useful for compiler optimisations, if we can add `const` in some places where IdLists are used.

Reply:
I'd be interested in seeing evidence of any difference in this case.